{"name": "test_create_user", "status": "failed", "statusDetails": {"message": "Failed: Ошибка валидации Json схемы: 200 is not of type 'string'\n\nFailed validating 'type' in schema['properties']['code']:\n    {'type': 'string'}\n\nOn instance['code']:\n    200", "trace": "self = <tests.test_user_operations.TestUserOperations object at 0x0000023207D6A930>\nuser_page = <pages.user_page.UserPage object at 0x000002320A704CE0>\nuser_data = <config.data.user_data object at 0x000002320A704FB0>\n\n    def test_create_user(self, user_page, user_data):\n        with allure.step(\"Создание пользователя\"):\n            response = user_page.create_user(user_data.__dict__)\n            assert response.status_code == 200, response.text\n    \n            with allure.step(\"Проверка ответа\"):\n                allure.attach(response.text, name=\"Текст ответа\", attachment_type=allure.attachment_type.JSON)\n                allure.attach(str(response.headers), name=\"Заголовки\", attachment_type=allure.attachment_type.JSON)\n    \n            if response.headers.get('Content-Type') == 'application/json':\n                r = response.json()\n                with allure.step(\"Проверка Json схемы ответа\"):\n                    try:\n>                       validate(instance=r, schema=response_create_user_schema)\n\ntest_user_operations.py:23: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\ninstance = {'code': 200, 'message': '59', 'type': 'unknown'}\nschema = {'properties': {'code': {'type': 'string'}, 'message': {'type': 'string'}, 'type': {'type': 'string'}}, 'required': ['code', 'type', 'message'], 'type': 'object'}\ncls = <class 'jsonschema.validators.Draft202012Validator'>, args = ()\nkwargs = {}\nvalidator = Draft202012Validator(schema={'properties': {'code': {'type': 'string'}, 'message': {'type': 'string'}, 'type': {'type': 'string'}}, 'required': ['code', 'type', 'message'], 'type': 'object'}, format_checker=None)\nerror = <ValidationError: \"200 is not of type 'string'\">\n\n    def validate(instance, schema, cls=None, *args, **kwargs):  # noqa: D417\n        \"\"\"\n        Validate an instance under the given schema.\n    \n            >>> validate([2, 3, 4], {\"maxItems\": 2})\n            Traceback (most recent call last):\n                ...\n            ValidationError: [2, 3, 4] is too long\n    \n        :func:`~jsonschema.validators.validate` will first verify that the\n        provided schema is itself valid, since not doing so can lead to less\n        obvious error messages and fail in less obvious or consistent ways.\n    \n        If you know you have a valid schema already, especially\n        if you intend to validate multiple instances with\n        the same schema, you likely would prefer using the\n        `jsonschema.protocols.Validator.validate` method directly on a\n        specific validator (e.g. ``Draft202012Validator.validate``).\n    \n    \n        Arguments:\n    \n            instance:\n    \n                The instance to validate\n    \n            schema:\n    \n                The schema to validate with\n    \n            cls (jsonschema.protocols.Validator):\n    \n                The class that will be used to validate the instance.\n    \n        If the ``cls`` argument is not provided, two things will happen\n        in accordance with the specification. First, if the schema has a\n        :kw:`$schema` keyword containing a known meta-schema [#]_ then the\n        proper validator will be used. The specification recommends that\n        all schemas contain :kw:`$schema` properties for this reason. If no\n        :kw:`$schema` property is found, the default validator class is the\n        latest released draft.\n    \n        Any other provided positional and keyword arguments will be passed\n        on when instantiating the ``cls``.\n    \n        Raises:\n    \n            `jsonschema.exceptions.ValidationError`:\n    \n                if the instance is invalid\n    \n            `jsonschema.exceptions.SchemaError`:\n    \n                if the schema itself is invalid\n    \n        .. rubric:: Footnotes\n        .. [#] known by a validator registered with\n            `jsonschema.validators.validates`\n    \n        \"\"\"\n        if cls is None:\n            cls = validator_for(schema)\n    \n        cls.check_schema(schema)\n        validator = cls(schema, *args, **kwargs)\n        error = exceptions.best_match(validator.iter_errors(instance))\n        if error is not None:\n>           raise error\nE           jsonschema.exceptions.ValidationError: 200 is not of type 'string'\nE           \nE           Failed validating 'type' in schema['properties']['code']:\nE               {'type': 'string'}\nE           \nE           On instance['code']:\nE               200\n\n..\\.venv\\Lib\\site-packages\\jsonschema\\validators.py:1332: ValidationError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <tests.test_user_operations.TestUserOperations object at 0x0000023207D6A930>\nuser_page = <pages.user_page.UserPage object at 0x000002320A704CE0>\nuser_data = <config.data.user_data object at 0x000002320A704FB0>\n\n    def test_create_user(self, user_page, user_data):\n        with allure.step(\"Создание пользователя\"):\n            response = user_page.create_user(user_data.__dict__)\n            assert response.status_code == 200, response.text\n    \n            with allure.step(\"Проверка ответа\"):\n                allure.attach(response.text, name=\"Текст ответа\", attachment_type=allure.attachment_type.JSON)\n                allure.attach(str(response.headers), name=\"Заголовки\", attachment_type=allure.attachment_type.JSON)\n    \n            if response.headers.get('Content-Type') == 'application/json':\n                r = response.json()\n                with allure.step(\"Проверка Json схемы ответа\"):\n                    try:\n                        validate(instance=r, schema=response_create_user_schema)\n                    except ValidationError as e:\n>                       pytest.fail(f\"Ошибка валидации Json схемы: {e}\")\nE                       Failed: Ошибка валидации Json схемы: 200 is not of type 'string'\nE                       \nE                       Failed validating 'type' in schema['properties']['code']:\nE                           {'type': 'string'}\nE                       \nE                       On instance['code']:\nE                           200\n\ntest_user_operations.py:25: Failed"}, "steps": [{"name": "Создание пользователя", "status": "failed", "statusDetails": {"message": "Failed: Ошибка валидации Json схемы: 200 is not of type 'string'\n\nFailed validating 'type' in schema['properties']['code']:\n    {'type': 'string'}\n\nOn instance['code']:\n    200\n", "trace": "  File \"C:\\Users\\bbuld\\PycharmProjects\\PetShop\\tests\\test_user_operations.py\", line 25, in test_create_user\n    pytest.fail(f\"Ошибка валидации Json схемы: {e}\")\n  File \"C:\\Users\\bbuld\\PycharmProjects\\PetShop\\.venv\\Lib\\site-packages\\_pytest\\outcomes.py\", line 177, in fail\n    raise Failed(msg=reason, pytrace=pytrace)\n"}, "steps": [{"name": "Проверка ответа", "status": "passed", "attachments": [{"name": "Текст ответа", "source": "82a014b7-7fd4-4165-a576-ffc7fa2841a3-attachment.json", "type": "application/json"}, {"name": "Заголовки", "source": "3425ff57-405a-4fdc-a353-eea3a6856f60-attachment.json", "type": "application/json"}], "start": 1719735386671, "stop": 1719735386672}, {"name": "Проверка Json схемы ответа", "status": "failed", "statusDetails": {"message": "Failed: Ошибка валидации Json схемы: 200 is not of type 'string'\n\nFailed validating 'type' in schema['properties']['code']:\n    {'type': 'string'}\n\nOn instance['code']:\n    200\n", "trace": "  File \"C:\\Users\\bbuld\\PycharmProjects\\PetShop\\tests\\test_user_operations.py\", line 25, in test_create_user\n    pytest.fail(f\"Ошибка валидации Json схемы: {e}\")\n  File \"C:\\Users\\bbuld\\PycharmProjects\\PetShop\\.venv\\Lib\\site-packages\\_pytest\\outcomes.py\", line 177, in fail\n    raise Failed(msg=reason, pytrace=pytrace)\n"}, "start": 1719735386672, "stop": 1719735386673}], "start": 1719735385552, "stop": 1719735386674}], "start": 1719735385552, "stop": 1719735386675, "uuid": "1c1cf2a7-c1a3-4153-be87-408b64bd6dcc", "historyId": "7cc3c00866496331b4cf80f932fcc455", "testCaseId": "7cc3c00866496331b4cf80f932fcc455", "fullName": "tests.test_user_operations.TestUserOperations#test_create_user", "labels": [{"name": "tag", "value": "@pytest.mark.usefixtures('user_page', 'user_data', 'created_user')"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_user_operations"}, {"name": "subSuite", "value": "TestUserOperations"}, {"name": "host", "value": "sanya"}, {"name": "thread", "value": "16148-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_user_operations"}]}